ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java

```diff
@@ -48,7 +48,9 @@ protected long getNextExecutingTime(TableRuntime tableRuntime) {
 
   @Override
   public void handleStatusChanged(TableRuntime tableRuntime, OptimizingStatus originalStatus) {
-    tryEvaluatingPendingInput(tableRuntime, loadTable(tableRuntime));
+    if (tableRuntime.getOptimizingStatus().equals(OptimizingStatus.IDLE)) {
```

Comment:
The parameter `originalStatus` doesn't work here? Should we follow the definition of the interface and compare the original status with the current status?

I think AMS will call this method in the following case:
PENDING -> IDLE(when clean up pending inputs): there is no need for a processing task, but we have to scan the file again in `OptimizingEvaluator`?
Do I understand correctly?

Reply:
Yes, I think you are right.

We can limit the evaluation process only if the status changed from processing to idle with codes like :
```
if (originalStatus != null && originalStatus.isProcessing() && tableRuntime.getOptimizingStatus().equals(OptimizingStatus.IDLE)) {

```

Reply:
agree with up floors

Reply:
It seems that status transformation is quite enumerable, and in my opinion, any status changed to `IDLE` could trigger evaluating, because `IDLE` status is the only status previous to evaluating(may be we shall define a status of evaluating in the future) and the condition is necessary and sufficient.

so I think the code is proper